department_app/service/utils.py

```diff
@@ -0,0 +1,98 @@
+"""interaction with databases"""
+
+from datetime import datetime
+
+from flask import abort
+
+from department_app.models import db
+
+
+def get_all(model):
+    """return all the record in the
+       table represented by model
+       if the model does not exist
+       raise ValueError if the table
+       is empty return empty list []
+
+       model: is a model of some table in the db
+    """
+    try:
+        return model.query.all()
+    except Exception:
+        raise ValueError('given Model "%s" is invalid' % model) from None
+
+
+def get_or_404(model, **criterion):
+    """get the record from the model
+       based on the given criterion
+       if the record or the given
+       criterion is invalid raise NotFound
+       exception, the same happens when
+       the invalid model is given
+
+       model: is a model of some table in the db
+       criterion: is a criterion on the function will filter
+                  the result (e.g id=1 or name="Management department")
+       """
+    try:
+        return model.query.filter_by(**criterion).first_or_404()
+    except Exception:
+        abort(404, 'The record was not found')
+
+
+def insert_into_db(record):
+    """insert record into the database, if record already exists
+       return BadRequest, if record is invalid raise BadRequest
+
+       record: is an object of the model class
+    """
+    if not isinstance(record, db.Model):
+        abort(400, 'ivalid record %s' % record)
+
+    try:
+        db.session.rollback()
```

Comment:
I suppose this rollback call is redundant

Reply:
done